packages/ruleset/src/utils/date-based-utils.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+/**
+ * This function takes a schema property name and uses a collection of regular
+ * expressions to heuristically determine if the property is a date or a date
+ * time property. It returns `true` if the name matches one of the expressions.
+ *
+ * @param {string} name the name of a schema property
+ * @returns a boolean value indicating that the property seems to be date-based
+ */
+function isDateBasedName(name) {
+  // Check for matching against certain patterns.
+  const dateBasedNamePatterns = [
+    // The name `created`.
+    new RegExp('^created$'),
```

Comment:
FWIW, I prefer native RegExp literal synax.

```suggestion
    /^created$/,
```

Reply:
I do, too - I initially made this change for a linter problem that seems to have been caused by something else (because I can't reproduce it). So, I'll go back to this 👍